utils/layout.tsx

```diff
@@ -1,16 +1,16 @@
 import { NextPage } from 'next';
-import { ReactElement, ReactNode } from 'react';
+import { FC, ReactNode } from 'react';
 
-export type Layout = (page: ReactElement) => ReactNode;
+export type Layout = FC<{ children: ReactNode }>;
 
 export type NextPageWithLayout<P = unknown, IP = P> = NextPage<P, IP> & {
   getLayout?: Layout;
 };
 
 export function getLayout<P, IP>(Component: NextPageWithLayout<P, IP>) {
-  return Component.getLayout ?? ((page) => page);
+  return Component.getLayout ?? (({ children }) => children);
 }
 
-export function setLayout(Component: NextPage, layout: Layout) {
-  (Component as NextPageWithLayout).getLayout = layout;
+export function setLayout(Component: NextPage, Layout: Layout) {
```

Comment:
Layout 파라미터에 어떤 레이아웃을 넣을 수 있을지 좀 더 쉽게 알 수 있도록 하면 좋을 것 같아요
그래서 enum 같은 걸로 레이아웃 종류를 미리 지정해놓고 쓰는 건 어떨까요?

```suggestion
type Layouts = 'HEADER' | 'EMPTY' | 'CUSTOM';
const LAYOUTS = {
  HEADER: HeaderLayout,
  // 생략
}

export function setLayout(Component: NextPage, Layout: Layouts, CustomLayout?: ReactNode) {
switch(Layout) {
case 'EMPTY':
return LAYOUTS['EMPTY']
// 생략
case 'CUSTOM':
return CustomLayout;
```

웬만하면 자주 쓰이는 레이아웃을 쓸 것이라 enum 안에 정의해서 쓰면 편할 거라 생각해요
근데 혹시나 기존에 정의한 레이아웃을 특정 페이지에서만 조금 변형해서 쓴다든지 하는 경우를 위해 커스텀 레이아웃을 옵션으로 받을 수 있게 해봤어요!